server/persistence/kube/read_client.go

```diff
@@ -58,36 +60,112 @@
 	objs *workspacesv1alpha1.WorkspaceList,
 	opts ...client.ListOption,
 ) error {
-	sbb := toolchainv1alpha1.SpaceBindingList{}
-	if err := c.backend.List(ctx, &sbb, &client.ListOptions{}); err != nil {
+	// retrieve workspaces visible to user
+	ww := workspacesv1alpha1.WorkspaceList{}
+	if err := c.fetchAllWorkspacesVisibileToUser(ctx, user, &ww); err != nil {
 		return err
 	}
 
-	if len(sbb.Items) == 0 {
-		return nil
-	}
+	rww := workspacesv1alpha1.WorkspaceList{}
+
+	// apply label selectors and transform workspaces
+	listOpts := client.ListOptions{}
+	listOpts.ApplyOptions(opts)
+	for _, w := range ww.Items {
+		// selection
+		if !matchesListOpts(&listOpts, w.GetLabels()) {
+			continue
+		}
 
-	for _, sb := range sbb.Items {
-		if sb.Spec.MasterUserRecord != user {
+		// transform
+		// override workspaces namespace with owner
+		ow, err := workspacesutil.GetOwner(&w)
```

Comment:
as we are using go 1.22 this should be a false positive. @sadlerap ideas on this?

Reply:
I think we're safe from this one, since we're running go v1.22.  We can probably exclude this error in our linter config and in gosec.

See also: https://github.com/konflux-workspaces/workspaces/security/code-scanning/9

Reply:
add exclusions in #57